apps/system/js/audio_channel_policy.js

```diff
@@ -0,0 +1,273 @@
+      system: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      ringer: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      telephony: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      notification: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      publicNotification: { activeAudioChannel: PLAY, newAudioChannel: PLAY }
+    },
+    publicNotification: {
+      normal: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      content: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      alarm: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      system: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      ringer: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      telephony: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      notification: { activeAudioChannel: PLAY, newAudioChannel: PLAY },
+      publicNotification: { activeAudioChannel: PLAY, newAudioChannel: PLAY }
+    }
+  };
+
+  /**
+   * AudioChannelPolicy provides policies to handle audio channels.
+   */
+  var AudioChannelPolicy = function() {};
+
+  BaseModule.create(AudioChannelPolicy, {
+    name: 'AudioChannelPolicy',
+    DEBUG: false,
+    /**
+     * Get the policies of 
+     * handling the new audio channel and active audio channels.
+     *
+     * @param {AudioChannelController} audioChannel
+     * A new channel you want to handel.
+     * @return {Object} The policies for active and new audio channels.
+     * The object is like below
+     * {
+     *   activeAudioChannels: [
+     *     { isAllowedToPlay: false,
+     *       isResumedWhenOtherEnds: true,
+     *       audioChannel: fm
+     *     },
+     *     {
+     *       isAllowedToPlay: true,
+     *       audioChannel: notification 
+     *     }
+     *   ],
+     *   newAudioChannel: {
+     *     isAllowedToPlay: true,
+     *     isNeededToFadeOut: true,
+     *     audioChannel: music 
+     *   }
+     * }
+     */
+    compete: function(newAudioChannel, activeAudioChannels, options) {
+      var policies = { newAudioChannel: null, activeAudioChannels: null };
+      var newAudioChannelName = newAudioChannel.channelName;
+      var isAllowedToPlayForNewAduioChannel = true;
+      var isNeededToFadeOutForNewAduioChannel = false;
+      if (activeAudioChannels.size) {
+        policies.activeAudioChannels = [];
+        activeAudioChannels.forEach((activeAudioChannel) => {
+          var activeAudioChannelName = activeAudioChannel.channelName;
+          var results = AUDIO_CHANNEL_COMPETITION_RESULES
+            [activeAudioChannelName][newAudioChannelName];
+          var isNeededToFadeOut = this._isNeededToFadeOutForActiveAudioChannel(
+              activeAudioChannelName, newAudioChannelName);
+          var isNeededToVibrate = this._isNeededToVibrateForActiveAudioChannel(
+              activeAudioChannelName, newAudioChannelName);
+          var isResumedWhenOtherEnds;
```

Comment:
isResumedWhenOtherEnds -> isNeededToResumeWhenOtherEnds to be consistent with the above 2 variables.